src/pcapi/core/users/models.py

```diff
@@ -113,36 +97,65 @@ class NotificationSubscriptions:
     marketing_email: bool = True
 
 
+# calculate date of latest birthday
+def _get_latest_birthday(birth_date: date) -> date:
+    """
+    Calculates the latest birthday of a given person.
+    :param birth_date: The person's birthday.
+    :return: The latest birthday's date.
+    """
+
+    today = date.today()
+
+    # handle february 29th
+    if birth_date.month == 2 and birth_date.day == 29:
+        try:
+            this_year_birthday = date(today.year, birth_date.month, birth_date.day)
+        except ValueError:
+            this_year_birthday = date(today.year, 2, 28)
+        try:
+            previous_year_birthday = date(today.year - 1, birth_date.month, birth_date.day)
+        except ValueError:
+            previous_year_birthday = date(today.year - 1, 2, 28)
+    else:
+        this_year_birthday = date(today.year, birth_date.month, birth_date.day)
+        previous_year_birthday = date(today.year - 1, birth_date.month, birth_date.day)
+
+    if birth_date.month > today.month or (birth_date.month == today.month and birth_date.day > today.day):
+        return previous_year_birthday
+    return this_year_birthday
```

Comment:
Si je comprends bien, on cherche à savoir si la date d'anniversaire est passée ? Est-ce qu'on ne pourrait pas écrire ça comme ça :

```suggestion
    if this_year_birthday <= today:
        return this_year_birthday
    return previous_year_birthday
```